src/System.Runtime/ref/System.Runtime.cs

```diff
@@ -6139,7 +6139,7 @@ public class TypeDelegator : System.Reflection.TypeInfo
     public delegate bool TypeFilter(Type m, Object filterCriteria);
     public abstract partial class TypeInfo : System.Type, System.Reflection.IReflectableType
     {
-        internal TypeInfo() { }
+        protected TypeInfo() { }
```

Comment:
It has to be done for the System.Reflection.Emit contract to compile.

So this may be a reason not to change the ancestry of those types. That's we need to figure out.